stl/inc/format

```diff
@@ -302,69 +418,76 @@ constexpr const _CharT* _Parse_format_specs(const _CharT* _Begin, const _CharT*
     return _Begin;
 }
 
-template <class _Ty, class _CharT = char>
-struct formatter;
-
-inline void _You_see_this_error_because_arg_id_is_out_of_range() noexcept {}
-
-template <class _CharT>
-class basic_format_parse_context {
-public:
-    using char_type      = _CharT;
-    using const_iterator = typename basic_string_view<_CharT>::const_iterator;
-    using iterator       = const_iterator;
-
-    constexpr explicit basic_format_parse_context(basic_string_view<_CharT> _Fmt, size_t _Num_args_ = 0) noexcept
-        : _Format_string(_Fmt), _Num_args(_Num_args_) {}
-    basic_format_parse_context(const basic_format_parse_context&) = delete;
-    basic_format_parse_context& operator=(const basic_format_parse_context&) = delete;
-
-    _NODISCARD constexpr const_iterator begin() const noexcept {
-        return _Format_string.begin();
-    }
-    _NODISCARD constexpr const_iterator end() const noexcept {
-        return _Format_string.end();
-    }
-    constexpr void advance_to(const const_iterator _It) {
-        _Adl_verify_range(_It, _Format_string.end());
-        const auto _Diff = static_cast<size_t>(_It._Unwrapped() - _Format_string._Unchecked_begin());
-        _Format_string.remove_prefix(_Diff);
+template <class _CharT, _Parse_replacement_field_callbacks<_CharT> _HandlerT>
+constexpr const _CharT* _Parse_replacement_field(const _CharT* _Begin, const _CharT* _End, _HandlerT&& _Handler) {
+    ++_Begin;
+    if (_Begin == _End) {
+        throw format_error("Invalid format string.");
     }
 
-    // While the standard presents an exposition only enum value for
-    // the indexing mode (manual, automatic, or unknown) we use _Next_arg_id to indicate it.
-    // _Next_arg_id == 0 means unknown
-    // _Next_arg_id > 0 means automatic
-    // _Next_arg_id == -1 means manual
-    constexpr size_t next_arg_id() {
-        if (_Next_arg_id == -1) {
-            throw format_error("Can not switch from manual to automatic indexing");
+    if (*_Begin == '}') {
+        // string was "{}", and we have a replacement field
+        _Handler._On_replacement_field(_Handler._Parse_context.next_arg_id(), _Begin);
+    } else if (*_Begin == '{') {
+        // string was "{{", so we have a literal "{" to print
+        _Handler._On_text(_Begin, _Begin + 1);
+    } else {
+        _Id_adapter<_CharT> _Adapter{_Handler._Parse_context};
+        _Begin     = _Parse_arg_id(_Begin, _End, _Adapter);
+        _CharT _Ch = _CharT{};
+        if (_Begin != _End) {
+            _Ch = *_Begin;
+        }

```

Comment:
I think that if `_Begin == _End` we will always end up in the throw error case.

At least for me It would be clearer if we would write it like that
```suggestion
        _CharT _Ch = _CharT{};
        if (_Begin != _End) {
            _Ch = *_Begin;
        }
```suggestion
        if (_Begin == _End) {
            throw format_error("Missing '}' in format string.");
        }
        
        
       const  _CharT _Ch = *_Begin;
```
```